compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/CodeAnalyzer.java

```diff
@@ -1279,6 +1280,18 @@ private boolean isValidTransactionBlock() {
                 !this.withinRetryBlock;
     }
 
+    private void validateMainFunction(BLangFunction funcNode) {
+        if (MAIN_FUNCTION_NAME.equals(funcNode.name.value)) {
```

Comment:
I think it is better to invert the if condition. WDYT?

Reply:
sorry, invert how?

Reply:
```java
if (!MAIN_FUNCTION_NAME.equals(funcNode.name.value)) {
    return;
}
if (!Symbols.isPublic(funcNode.symbol)) {
    this.dlog.error(funcNode.pos, DiagnosticCode.MAIN_CANNOT_BE_NON_PUBLIC);
}
if (!(funcNode.symbol.retType.tag == TypeTags.NIL || funcNode.symbol.retType.tag == TypeTags.INT)) {
    this.dlog.error(funcNode.returnTypeNode.pos, DiagnosticCode.INVALID_RETURN_WITH_MAIN,
                    funcNode.symbol.retType);
}
```

It is better to return from a method as early as possible.